sql/server.go

```diff
@@ -154,82 +168,73 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
-func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
-
-	// Pick up current session state.
-	// The request user is validated in ServeHTTP. Even in insecure mode,
-	// it is guaranteed not to be empty.
-	planMaker := planner{user: req.GetUser()}
-	defer func() {
-		// Update session state even if an error occurs.
-		if bytes, err := gogoproto.Marshal(&planMaker.session); err != nil {
-			// Marshaling a `Session` never errors (known from reading the code).
-			panic(err)
-		} else {
-			resp.Session = bytes
-		}
-	}()
+func (s *Server) exec(req driver.Request, planMaker *planner) driver.Response {
+	var resp driver.Response
 
-	if req.Session != nil {
-		// TODO(tschottdorf) will have to validate the Session information (for
-		// instance, whether access to the stored database is permitted).
-		if err = gogoproto.Unmarshal(req.Session, &planMaker.session); err != nil {
-			return
-		}
-	}
-	var stmts parser.StatementList
-	if stmts, err = parser.Parse(req.Sql); err != nil {
-		return
+	// TODO(vivek): This should parse each statement independently, report parse
```

Comment:
How would we parse each statement independently when they are sent in a single string? We can't reasonably ask the client to break the multiple statements into multiple strings as that involves parsing the SQL. And trying to execute some of the statements when a parse error occurred in one of them sounds like a recipe for errors. I'd nuke this TODO unless you have some reasonable answers here.